### PR TITLE
Chapter 5: typos + explanation on lambda-lifting + maximal insertion on Monoid.law

### PR DESCRIPTION
**Pull request by gallais**

### CODE REVIEW

File: tex/chTypeInference.tex [hunk 1]
```diff
@@ -1421,7 +1421,12 @@ \subsection{Assumptions of a bigop lemma}\label{sec:bigoplemmas}
 canonical structures) to the three properties of monoids.
 
 We can then us this interface as a parameter for a bunch of lemmas,
-describing the theory shared by  its instances.
+describing the theory shared by  its instances. The \C{(Monoid.law idx)}
+type annotation may seem puzzling at first; it combines various mechanism
```
Comment (anton-trunov) [inline tex/chTypeInference.tex]: "mechanism" -> "mechanismS"

**Comment by gallais** [comment]
Thanks! (I push-forced to keep a clean history. Looks like it got rid of your comment.)

**Comment by anton-trunov** [comment]
I still can see it, it's just behind the "Show outdated" link.

**Comment by anton-trunov** [comment]
If the book is aiming at mathematicians (as opposed to cs people) then I think "lambda-lifted" needs to be explained too.
By the way, "maximally inserted" occurs earlier in the book, but it isn't explained.

**Comment by gares** [comment]
Thanks for the fixes!  I agree with @anton-trunov  that lambda-lift is bit too CS oriented, but I see your point: the sentence needs to be more clear.

**Comment by gares** [comment]
Thanks!

**Pull request merged**